receiver/httpcheckreceiver/metadata.yaml

```diff
@@ -0,0 +1,24 @@
+name: httpcheckreceiver
+
+resource_attributes:
+
+attributes:
+  http.url:
+    description: Full HTTP request URL.
+    type: string
+  http.status_code:
+    description: HTTP response status code
+    type: int
+  http.method:
+    description: HTTP request method
+    type: string
+
+metrics:
+  httpcheck.success:
```

Comment:
Is it worth considering any other representations of this metric? One thought is that if this were `httpcheck.status` where the value is the status code, then you wouldn't have to interpret what success means. Success, failure, etc could be determined on the query side.

Reply:
That's correct, the representation could be the status code itself (which is currently an attribute). I don't have a strong preference one way or the other. The success/failure metric is similar to `probe_success` emitted by https://github.com/prometheus/blackbox_exporter. It does mean that this receiver will need to have a way for users to configure success/failure.

Reply:
Out of curiosity, do you expect users to check the method attribute? e.g. probe both GET and POST in the same service?

Reply:
i wouldn't expect someone to want to hit the same endpoint with different methods. i would expect consumers of the telemetry to want to be able to filter responses by the method though.

Reply:
I think it would make a lot of sense to keep the status code as an attribute, but also to change the meaning of the value from "success" to "a single observation of this status code". 

You would expect subsequent observations to look like this:
```
{ "url": "/foo", "method": "GET", "code": 200, "value": 1, "time": "t0" }
{ "url": "/foo", "method": "GET", "code": 200, "value": 1, "time": "t1" }
{ "url": "/foo", "method": "GET", "code": 404, "value": 1, "time": "t2" } // outage
{ "url": "/foo", "method": "GET", "code": 200, "value": 1, "time": "t3" }
```

See my other comment for more on why I think this model is useful for summarization.

Reply:
See my suggestion to make this a 0-1 metric similar to or perhaps specified to create an OpenMetrics `StateSet`. @dashpole note the connection to #1712 and #2409.

Reply:
I've reviewed your other comment, and I think this would be a great way to represent this sort of information!  A unit seems like the best place I can think of for putting that information, and it makes sense in contexts other than Prometheus.

Reply:
Seems like we could take a similar approach for info metrics as well.

Reply:
Sounds good!